ostd/src/mm/frame/meta.rs

```diff
@@ -69,19 +71,21 @@ pub(in crate::mm) struct MetaSlot {
     ///
     /// It is placed at the beginning of a slot because:
     ///  - the implementation can simply cast a `*const MetaSlot`
-    ///    to a `*const PageMeta` for manipulation;
+    ///    to a `*const FrameMeta` for manipulation;
     ///  - if the metadata need special alignment, we can provide
     ///    at most `PAGE_METADATA_ALIGN` bytes of alignment;
     ///  - the subsequent fields can utilize the padding of the
     ///    reference count to save space.
     storage: UnsafeCell<[u8; PAGE_METADATA_MAX_SIZE]>,
+    /// Whether the page is untyped.
+    pub(super) is_untyped: UnsafeCell<bool>,
```

Comment:
Oh, I missed this comment (GitHub doesn't seem to automatically load new reviews? It _does_ automatically load new comments, though).

https://github.com/asterinas/asterinas/pull/1752#discussion_r1897187578

> I think it doesn't involve synchronization since we only sync on the reference count. Why should it be a `AtomicBool` btw?

Using `UnsafeCell` should be fine, just somehow not very convenient.
 - Ideally, we should merge all three `UnsafeCell`s into one, because they follow exactly the same principle to get the immutable/mutable reference. But we cannot do this, because `vtable_ptr` has to be placed after `ref_count`, but two other `UnsafeCell`s are before `ref_count`.
 - So I'm just thinking about whether using `AtomicBool` can simplify the code. It's true that we don't really need to do the synchronization on this boolean variable, but reading/writing a boolean can be atomic without any cost (in most common architectures?). So we don't have to write unsafe code to operate on this variable.

However, I fully admit that it's a strong argument, so feel free to keep the current form if you don't agree. I think your point makes sense as well.